java/ql/src/Telemetry/AutomodelApplicationModeExtractNegativeExamples.ql

```diff
@@ -18,7 +18,7 @@ private import AutomodelSharedUtil
  */
 bindingset[limit]
 Endpoint getSampleForCharacteristic(EndpointCharacteristic c, int limit) {
-  exists(int n |
+  exists(int n, int num_endpoints | num_endpoints = count(Endpoint e | c.appliesToEndpoint(e)) |
```

Comment:
Apologies if this not the right place but could we add some comment/javadoc detailing our sampling strategy here? The diversity/quality of the provided labelled examples are often a crucial part of a model's performance - not sure whether that will be the case with LLM but it'd be good to document our approach anyway.

Reply:
there's a comment a few lines below — have you seen it?

Reply:
do you mean `// we order the endpoints by location, but (to avoid bias) we select the indices semi-randomly` ?

Reply:
Yes — seems like it describes the behaviour. But I'm happy to add more text if you have a suggestion! :+1:

Reply:
I think some sort English description detailing what we're trying to do, why and how would be great. For instance what does `semi-randomly` mean? Are we trying to sample the different types of negative samples evenly?

Reply:
I'll rewrite it to be more clear what we're doing and why. (I wanted to rewrite it anyway to get rid of the randomness comment.)

Reply:
Done in [ea5c364](https://github.com/github/codeql/pull/13239/commits/ea5c36491b3581e40ea56ca1a946ad93eefeced8).

Reply:
Awesome — thanks, Taus and Jean